### DISCUSSION TITLE
access to "prediction without track identity"

### DISCUSSION THREAD

**Discussion by adpala** [talmolab/sleap#1171] [Help!] [answered]
Hi,

Is it possible to get instances labeled as "prediction without track identity" from an ".slp" file?

At the moment I just convert from the .slp to the analysis hdf5 file like this:
`sleap-convert --output OUTPUT_FILE --video INPUT_VIDEO --format analysis OUTPUT_FILE.slp`

But I haven't found where these instances are stored. I'm interested in these instances, since they correctly fit the animal in those video frames, although for some reason (not the topic of the question) these are not assigned a track.

**Comment by roomrys** [comment, marked as answer]
Hi @adpala,

Only instances with track identities are saved in the analysis hdf5. As you stated, you will need to navigate within the slp file directly. We have a [data structures notebook](https://sleap.ai/notebooks/Data_structures.html) which gives a few examples for how to navigate the data in your slp.

Basically each slp file is contained within a `Labels` object.
https://github.com/talmolab/sleap/blob/57ee08a7490aa950f61b14f80fa8e2d372a2ca5c/sleap/io/dataset.py#L390-L426

The `Labels` object contains a list of `LabeledFrame`s (accessed through `Labels.labeled_frames`). 
https://github.com/talmolab/sleap/blob/57ee08a7490aa950f61b14f80fa8e2d372a2ca5c/sleap/instance.py#L1282-L1292

Each `LabeledFrame` contains a list of `PredictedInstance`s (accessed through `LabeledFrame.predicted_instances`). 
https://github.com/talmolab/sleap/blob/57ee08a7490aa950f61b14f80fa8e2d372a2ca5c/sleap/instance.py#L1431-L1434

Each `PredictedInstance` will have a `Track` (accessed through `PredictedInstance.track`). If the `Track` is `None`, then the `PredictedInstance` does not have a track.
https://github.com/talmolab/sleap/blob/57ee08a7490aa950f61b14f80fa8e2d372a2ca5c/sleap/instance.py#L344-L367

An (untested) MVP for your use case:
```python
import sleap

ds = "path/to/slp"
labels = sleap.load_file(ds)

lfs = labels.labeled_frames
instances_without_tracks = [inst for inst in lfs.predicted_instances if inst.track is None]
```

Let us know if you need any extra assistance!

Thanks,
Liezl

**Reply by adpala** [reply]
Thanks a lot Liezl, that was a great answer!
